app/src/components/ActionsPanel.js

```diff
@@ -1,137 +1,157 @@
+          onAction={onAction}
+          outsideToken={outsideToken}
+          wrappedToken={wrappedToken}
         />
       </SidePanel>
     )
   }
 )
 
-class WrapTokensPanelContent extends React.PureComponent {
-  static defaultProps = {
-    onAction: () => {},
-  }
-  state = {
-    ...initialState,
-  }
-  componentWillReceiveProps({ panelOpened }) {
-    if (panelOpened && !this.props.panelOpened) {
-      // setTimeout is needed as a small hack to wait until the input's on
-      // screen until we call focus
-      this.amountInput && setTimeout(() => this.amountInput.focus(), 0)
-    } else if (!panelOpened && this.props.panelOpened) {
-      // Finished closing the panel, so reset its state
-      this.setState({ ...initialState })
-    }
-  }
-  handleAmountChange = event => {
-    this.setState({ amount: event.target.value })
-  }
+function WrapTokensPanelContent({
+  action,
+  info,
+  onAction,
+  outsideToken,
+  wrappedToken,
+}) {
+  const theme = useTheme()
 
-  handleSubmit = event => {
-    event.preventDefault()
-    this.props.onAction(this.state.amount)
-  }
-  render() {
-    const { amount } = this.state
+  const [amount, setAmount] = useState('')
+  const tokenInputRef = useSidePanelFocusOnReady()
 
-    return (
-      <div>
-        <form
-          css={`
-            margin-top: ${3 * GU}px;
-          `}
-          onSubmit={this.handleSubmit}
-        >
+  const tokenData = useMemo(
+    () => (action === 'Wrap' ? outsideToken : wrappedToken),
+    [action, outsideToken, wrappedToken]
+  )
+  const handleAmountChange = useCallback(event => {
+    setAmount(event.target.value)
+  }, [])
+  const handleSubmit = useCallback(
+    event => {
+      event.preventDefault()
+
+      onAction(toDecimals(amount.trim(), tokenData.numDecimals))
+    },
+    [amount, onAction, tokenData]
+  )
+
+  const tokenStep = fromDecimals(
+    '1',
+    Math.min(MAX_INPUT_DECIMAL_BASE, tokenData.numDecimals)
+  )
+
+  return (
+    <form
+      css={`
+        margin-top: ${3 * GU}px;
+      `}
+      onSubmit={handleSubmit}
+    >
+      <div
+        css={`
+          margin-bottom: ${3 * GU}px;
+        `}
+      >
+        <Info>{info}</Info>
+      </div>
+      <Field label={action === 'Wrap' ? 'Amount' : 'Wrapped token amount'}>
+        <div css="display: flex">
+          <TextInput
+            ref={tokenInputRef}
+            type="number"
+            value={amount}
+            min={tokenStep}
+            step={tokenStep}
+            onChange={handleAmountChange}
+            adornment={
+              action === 'Wrap' ? outsideToken.symbol : wrappedToken.symbol
+            }
+            adornmentPosition="end"
+            adornmentSettings={{
+              width: 60,
+              padding: 8,
+            }}
+            required
+            wide
+            css={`
+              &::-webkit-inner-spin-button,
+              &::-webkit-outer-spin-button {
+                -webkit-appearance: none;
```

Comment:
This hides the arrow keys in some browsers, as it can look somewhat odd (this is something we can discuss with Pierre when he's back):

<img width="421" alt="Screen Shot 2019-12-03 at 4 26 30 PM" src="https://user-images.githubusercontent.com/4166642/70064403-c54de080-15e9-11ea-9462-ecedbaf3ba05.png">